R/zchunk_LA1231.elec_tech.R

```diff
@@ -37,59 +38,153 @@ module_energy_LA1231.elec_tech_DISABLED <- function(command, ...) {
+    L1231.eff_R_elec_gas_Yh_gathered %>%
+      select(GCAM_region_ID,sector,fuel, year,share_tech1) %>%
+      full_join(filter(L123.in_EJ_R_elec_F_Yh,fuel=="gas"),by=c("GCAM_region_ID","sector","fuel", "year")) %>%
+      rename(in_EJ=value) %>%
+      mutate(in_EJ_tech1=in_EJ*share_tech1) %>%
+      mutate(in_EJ_tech2=in_EJ-in_EJ_tech1) -> L1231.in_EJ_R_elec_gas_Yh_gathered
+
+    # Combined gas technologies (inputs) in long format
+    L1231.in_EJ_R_elec_gas_Yh_gathered %>%
+      select(-in_EJ,-share_tech1,-in_EJ_tech2) %>%
+      mutate(technology="gas (steam/CT)") %>%
+      rename(value=in_EJ_tech1) -> L1231.eff_R_elec_gas_tech1_Yh_gathered
+
+    L1231.in_EJ_R_elec_gas_Yh_gathered %>%
+      select(-in_EJ,-share_tech1,-in_EJ_tech1) %>%
+      mutate(technology="gas (CC)") %>%
+      rename(value=in_EJ_tech2) %>%
+      bind_rows(L1231.eff_R_elec_gas_tech1_Yh_gathered) %>%
+      select(GCAM_region_ID, sector,fuel,technology,year,value) -> L1231.in_EJ_R_elec_gas_tech_Yh
+
+    # Perform computations for technologies other than those for gas.
+    # All other (non-gas) technologies are not disaggregated further (only one tech per fuel type)
+
+    # Obtain efficiencies for technologies (other than gas technologies)
+    L123.eff_R_elec_F_Yh %>%
+      filter(fuel!="gas") %>%
+      left_join(select(calibrated_techs,sector,fuel,technology),by=c("sector" ,"fuel")) %>%
+      rename(efficiency=value) %>%
+      bind_rows(L1231.eff_R_elec_gas_tech_Yh) %>%
+      select(GCAM_region_ID,sector,fuel,technology,year,efficiency) -> L1231.eff_R_elec_F_tech_Yh
+
+    # Obtain inputs for technologies (other than gas technologies) and then create final input output by binding rows with the
+    # tibble that considers gas technologies only.
+    L123.in_EJ_R_elec_F_Yh %>%
+      filter(fuel!="gas") %>%
+      left_join(select(calibrated_techs,sector,fuel,technology),by=c("sector" ,"fuel")) %>%
+      bind_rows(L1231.in_EJ_R_elec_gas_tech_Yh) %>%
+      select(GCAM_region_ID,sector,fuel,technology,year,value) -> L1231.in_EJ_R_elec_F_tech_Yh
+
+    #Calculate output as input * efficiency
+    #This tibble only includes technologies with modeled efficiencies
+    L1231.eff_R_elec_F_tech_Yh %>%
+      left_join(L1231.in_EJ_R_elec_F_tech_Yh,by=c("GCAM_region_ID","sector","fuel","technology","year" )) %>%
+      mutate(output=efficiency*value) %>%
+      select(-value,-efficiency) -> L1231.out_EJ_R_elec_Fin_tech_Yh
+
+    # Rename efficiecy as value for final output in long format
+    L1231.eff_R_elec_F_tech_Yh %>%
+      rename(value=efficiency) -> L1231.eff_R_elec_F_tech_Yh
+
+    # Combine output with technologies modeled by output only (e.g. nuclear, hydro, renewables)
+    # and combined all outpues for all fuels and technologies to create final tibble in long format
+    L123.out_EJ_R_elec_F_Yh %>%
+     anti_join(L1231.out_EJ_R_elec_Fin_tech_Yh, by=c("GCAM_region_ID", "sector","fuel","year")) %>%
```

Comment:
Again, would be useful to say _why_ you're using an `anti_join` here. What's going on?